docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,359 @@
+  - `mediaType`: media type of the referrer
+  - `size`: referrer file size in bytes
+  - `digest`: digest of the referrer
+  - `artifactType`: artifact type of a referrer
+  - `annotations`: contains arbitrary metadata in a referrer
+
+See an example:
+
+```bash
+oras discover localhost:5000/hello:v1 --format json
+```
+
+```json
+{
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:0db683b656132cede5360f42bc52541f3386b30ce685e6e63ff93ced54423fb8",
+      "size": 964,
+      "annotations": {
+        "org.opencontainers.image.created": "2023-12-14T13:48:32Z"
+      },
+      "artifactType": "application/vnd.oci.empty.v1+json"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:476d43120d3799fa76d7706f741beca73f5ff4149c8b6db3bd516a73d4a82fc1",
+      "size": 728,
+      "annotations": {
+        "io.cncf.notary.x509chain.thumbprint#S256": "[\"792265ec6b22f0a87c7b3d980319d51a76a382de1b7a47bd877bb4e5a9beb637\"]",
+        "org.opencontainers.image.created": "2023-12-14T14:41:56Z"
+      },
+      "artifactType": "application/vnd.cncf.notary.signature"
+    }
+  ]
+}
+```
+
+> [!NOTE]
+> The `--format` flag will replace the existing `--output` flag. The `--output` will be marked as "deprecated" in ORAS v1.2.0 and will be removed in the future releases. 
+
+## Feature status
+
+This feature is marked as experimental in its first iteration since it is still in development and is available in a part of ORAS CLI commands.
+
+## FAQ
+
+**Q:** Why choose to use `--format` flag to enable JSON formatted output instead of extending the existing `--output` flag?
+**A:** ORAS follows [GNU](https://www.gnu.org/prep/standards/html_node/Option-Table.html#Option-Table) design principles. ORAS uses `--output` to specify a file or directory content should be created within and `--format` to format the output into JSON or using the given Go template. Popular tools, like Docker, Podman, and Skopeo also follow this design principle within their formatted output feature.
+
+**Q:** Why ORAS chooses [Go template](https://pkg.go.dev/text/template)?
+**A:** Go template is a powerful method to allow users to manipulate and customize output you want. It provides access to data objects and additional functions that are passed into the template engine programmatically. It also has some useful libraries that have strong functions for Go’s template language to manipulate the output data, such as [Sprig](https://masterminds.github.io/sprig/). The basic usage of Go template functions are easy to use.
+
+**Q:** What is the difference of prettified JSON and raw JSON?
+**A:** In the context of ORAS output, raw JSON means that `oras` outputs the JSON object as it is returned by the server, while prettified JSON means oras outputs the modified JSON objects with whitespace adjusted, such as formatted with human-friendly indentations and newlines.
```

Comment:
This QA seems to be unnecessary now.

Reply:
Make sense. Removed this QA as there is no "toRawJson" function in the examples